src/sage/interfaces/sage0.py

```diff
@@ -159,7 +159,8 @@ def __init__(self,
         if python:
             command = 'python -u'
             prompt = re.compile(b'>>> ')
-            init_code.append('from sage.all import *')
+            environment = 'sage.all'
```

Comment:
This seems like an unrelated and unnecessary change

Reply:
It is done because otherwise our linter (relint) complains about this line!

Reply:
I don't think it's a good idea to change the runtime behavior (although the impact is minimal here) to satisfy a linter. Is there not something like `relint-ignore-nextline` to ignore errors? Or improve the regex to not match in strings... Anyway, this is not part of this ticket...

Reply:
It is part of this PR because this PR takes care of `sage.interfaces`.
The follow up #35110 turns the relint warnings into errors again, so it has to be fixed.
No, there's no such thing as a "noqa" annotation for relint.

Reply:
> The follow up https://github.com/sagemath/sage/pull/35110 turns the relint warnings into errors again, so it has to be fixed.

Then I propose you fix it there.

Reply:
No thanks.

Reply:
That this instance of an "import" is found by the linter, is a bug in the linter and should be fixed accordingly. Since relint parses complete files, you can add a check that the line before does not contain a magic no-op comment, or fix the bug in another way (e.g. check that the import is not in a string, or at the beginning of a line). There are multiple ways to fix this properly.

Reply:
We need to progress here one way or another. Is it realistic to fix `relint` so that such a workaround is not needed?

Reply:
The linting rule is **designed** to flag all hardcoded namespace `.all` imports. No, not just at the beginning of a line, and no, also within strings and within doctests. There's no bug.

Reply:
1) Docstrings are strings.
2) These two instances are not false positives. All imports from `sage.all` need to be updated in some way. Above, in `sage.interfaces.sage0`, this defines a pexpect interface to sage itself; and the use of `sage.all` there will not work in a modularized setting. The solution will be to make the evaluation environment (module) configurable so it becomes suitable for modularized deployments. The change here, introducing a variable `environment` is a meaningful step into this direction; it is along the lines of the modularization changes to the doctester https://github.com/sagemath/sage/blob/develop/src/bin/sage-runtests#L60. 
(The full solution is, of course, out of the scope of this PR.)

Reply:
It is very simple: The linter has made me aware of it, and now I am aware.